### ISSUE TITLE
Cannot access remote files from fed sharing if the certificate changes

### ISSUE THREAD

**Issue by skshetry** [owncloud/core#35518]
**TLDR**: A user cannot access previously received folders/files from federation sharing, if the certificate changes, or receive any new shares. But, a new user can share/receive new files/folders. Or, if you delete `data/<user>/files_external/rootcerts.crt`, you can also access them gradually.

## Steps to reproduce
0. Extract ownCloud 10.2.0 release tarball into two folder. I copied as `www/release/ten2a` (**LOCAL** server) and `www/release/ten2b` (**REMOTE** server).
  `www` is `DocumentRoot` for me.
1. Create a new certificate if you haven't. `sudo openssl req -new -x509 -days 365 -nodes -out /etc/apache2/ssl/owncloud.pem -newkey rsa:2048 -keyout /etc/apache2/ssl/owncloud.key`

2. cd into root of the both servers. And, run the following commands (or, setup via webUI).
```
mkdir -p data apps-external config
sudo chown www-data:www-data -R apps-external data config .htaccess
sudo -u www-data ./occ maintenance:install --database-name=oc_release_$(basename $PWD) --database=mysql --database-user=root --database-pass root --admin-pass admin
sudo -u www-data ./occ config:system:set htaccess.RewriteBase --value /release/$(basename $PWD)
sudo -u www-data ./occ maintenance:update:htaccess
sudo -u www-data ./occ security:certificates:import /etc/apache2/ssl/owncloud.pem
```
3. Open REMOTE server with `admin`.
4. Open LOCAL server with `admin` (just to copy skeleton files).
5. Share `Photos` folder from REMOTE server to local server (`admin` to `admin` user).
6. Share `Documents` folder from REMOTE server to LOCAL server (`admin` to `admin` user).
6. Accept both shares from LOCAL server.
7. Notice the `Photos (2)`. Also check the `Photos (2)` is accessible and have same contents on both servers. But, don't check `Documents (2)`.
8. Regenerate new certificate (refer step 1.)
9. Remove old certificate and re-import new certificates on both servers.
10. Reload the server (if *apache*, `sudo systemctl restart apache2`).
11. The `Photos (2)` folder is accessible. The `Documents (2)` folder when opened, shows `You don't have permission to view ...`, and then grays-out.
	Opening it now will download a file instead, with content `This is the WebDAV interface. It can only be accessed by WebDAV clients such as the ownCloud desktop sync client.`.
12. In REMOTE server, Create `New folder` and upload a new file on that folder.
13. Share `New folder` folder from REMOTE server to LOCAL server (`admin` to `admin` user).
14. Accept share from LOCAL server.
15. Notice the `New folder`. The folder appears greyed out. And, cannot be accessed through webUI. Same as in step 11 for `Documents (2)`.
16. Curl for `curl -k "https://admin:admin@localhost/release/ten2a/remote.php/dav/files/admin/Documents%20(2)"`:
	```xml
	<?xml version="1.0"?>
	 <d:multistatus xmlns:d="DAV:" xmlns:s="http://sabredav.org/ns" xmlns:oc="http://owncloud.org/ns">
	   <d:response>
		 <d:href>/release/ten2a/remote.php/dav/files/admin/Documents%20(2)</d:href>
		 <d:propstat>
		   <d:prop>
			 <d:getlastmodified>Thu, 13 Jun 2019 10:47:40 GMT</d:getlastmodified>
			 <d:getcontentlength>0</d:getcontentlength>
			 <d:resourcetype/>
			 <d:getetag>"5d0229cc4814a"</d:getetag>
			 <d:getcontenttype/>
		   </d:prop>
		   <d:status>HTTP/1.1 200 OK</d:status>
		 </d:propstat>
	   </d:response>
	 </d:multistatus>
	```
	and, for `curl -k "https://admin:admin@localhost/release/ten2a/remote.php/dav/files/admin/Documents%20(2)/Example.odt" -X PROPFIND`:
	```xml
	<?xml version="1.0"?>
	<d:multistatus xmlns:d="DAV:" xmlns:s="http://sabredav.org/ns" xmlns:oc="http://owncloud.org/ns">
	  <d:response>
		<d:href>/release/ten2a/remote.php/dav/files/admin/Documents%20(2)/Example.odt</d:href>
		<d:propstat>
		  <d:prop>
			<d:getlastmodified>Thu, 13 Jun 2019 10:52:23 GMT</d:getlastmodified>
			<d:getcontentlength>0</d:getcontentlength>
			<d:resourcetype/>
			<d:getetag>"5d022ae7905f3"</d:getetag>
			<d:getcontenttype/>
		  </d:prop>
		  <d:status>HTTP/1.1 200 OK</d:status>
		</d:propstat>
	  </d:response>
	</d:multistatus>
	```
	Curl for `curl -k "https://admin:admin@localhost/release/ten2a/remote.php/dav/files/admin/New%20folder" -X PROPFIND`:
	```xml
	<?xml version="1.0"?>
    <d:multistatus xmlns:d="DAV:" xmlns:s="http://sabredav.org/ns" xmlns:oc="http://owncloud.org/ns">
      <d:response>
        <d:href>/release/ten2a/remote.php/dav/files/admin/New%20folder</d:href>
        <d:propstat>
          <d:prop>
            <d:getlastmodified>Thu, 13 Jun 2019 10:51:07 GMT</d:getlastmodified>
            <d:getcontentlength>0</d:getcontentlength>
            <d:resourcetype/>
            <d:getetag>"5d022a9b5c318"</d:getetag>
            <d:getcontenttype/>
          </d:prop>
          <d:status>HTTP/1.1 200 OK</d:status>
        </d:propstat>
      </d:response>
    </d:multistatus>
	```

#### Deleting root certificates and rechecking the folders/file for accessibility
17. Delete root certificate of admin user on LOCAL server. `sudo rm -rf data/admin/files_external/rootcerts.crt`.
18. The folder that are from external shares are still not accessible. Try to open any one of it, and then reload.
19. All the shares are accessible. If any files/folder are not accessible, try to open it and reload. Then, it should be accessible.

#### Deleting root certificates and resharing
*Note: The step follows after step 17 to realize step 22 completely.*

20. Delete root certificate of admin user on LOCAL server. `sudo rm -rf data/admin/files_external/rootcerts.crt`.
21. Create a new folder, share the folder from REMOTE server to local server (`admin` to `admin` user), and accept from LOCAL server.
22. The folder is accessible. Some of the times, all previously shared files were accessible too. If not, check above step 17.

#### Sharing with a new user
23. Create a new user in LOCAL server.
24. Open LOCAL server (ie. `ten2a` here) with the user. (just to copy skeleton files).
25. Share `Photos` folder from REMOTE server to local server (`admin` to `<new>` user).
26. Accept the share from LOCAL server.
27. The folder is accessible for `<new>`.

### Screenshots
1. Grayed out folder.
![Screenshot from 2019-06-13 16-52-09](https://user-images.githubusercontent.com/18718008/59427940-9b1a2380-8dfb-11e9-8363-d20ffa53fe34.png)

**Comment by micbar** [comment]
@skshetry Could be related to the trusted_servers mechanism

**Comment by individual-it** [comment]
need to retest with 10.1 to see if its a regression

**Comment by skshetry** [comment]
@micbar, I haven't checked with trusted_servers. Will try.

**Comment by skshetry** [comment]
I get exactly the same results on `v10.0.10`.

**Comment by skshetry** [comment]
I also get the same result when:

1. Add to trusted server on LOCAL server.
2. Perform all above actions.

And, when I:
1. Add to trusted server on LOCAL server.
2. Automatically accept shares.
3. Perform all above actions.

**Comment by DeepDiver1975** [comment]
🤖 Automated issue triage by Claude Code review agent.

This issue was opened before 2020 and has had no activity since then, so we are closing it as outdated. If this is still relevant on a current ownCloud release, please **reopen** (or open a fresh issue) with up-to-date details — thank you!
